### ISSUE TITLE
Questions about the interpretation of ALE results

### ISSUE THREAD

**Issue by wanxn518** [ssolo/ALE#42]
Hello, ALE team.

Thank you for the great software you have developed, it is very helpful for our research, I am new to phylogenetic problems and have encountered some confusing issues, I sincerely hope that you can give me some suggestions

For the ALEml_undated results I used branchwise_number_of_events.py to count the number of duplicates, transfers, losses, and origins for each branch, and in the first row of the image below, the branch number is shown as 118 for this branch, what does it represent on the species tree? How can I understand the meaning of this number? How can I visualise the species tree using itol, and how can I map it to the branches of the species tree? My impression. The phylogenetic tree has a node id and a tip id, but I'm not sure it represents the tip id

Can you give me any help? Sincere thanks in advance!

Best,
xnw

![image](https://github.com/ssolo/ALE/assets/97587984/8c01a5c1-c3f0-4900-96d2-a50c5d400ac7)

**Comment by zzlzef** [comment]
朋友你好，我最近也在尝试这个软件还有和它类似的AleRax，有个问题不太明白，就是我针对多个物种进行分析，每个物种都有很多个基因组，在进行物种树和基因树构建的时候，是应该每个物种都只选一个代表基因组进行分析，还是应该把所有物种的基因组都放在一起分析呢？

**Comment by ssolo** [comment]
Sajnos csak angol nyelvű kommentekre tudunk válaszolni.

**Comment by ssolo** [comment]
The beinging of the uml_recs file has a Newick string with the ids.

**Comment by zzlzef** [comment]
> The beinging of the uml_recs file has a Newick string with the ids.

Hello, this is a very interesting piece of software. I have a question that I’m a bit confused about. I’m analyzing multiple species, and each species includes many genomes. When constructing the species tree and gene trees, should I select only one representative genome per species for the analysis, or should I include all genomes from all species in the analysis together?

**Comment by ssolo** [comment]
Hi, that is not a software question, I don't know what the research question is or what hypothesis you are trying to test.

**Comment by zzlzef** [comment]
> Hi, that is not a software question, I don't know what the research question is or what hypothesis you are trying to test.

Hi, thanks. ALE reconciles a sample of gene trees with a species tree. I’m analyzing multiple species, and each species includes many genomes. when i construct the species tree, should I select only one representative genome per species to construct the species tree?

**Comment by ssolo** [comment]
Hi, yes, I know what ALE does. I'm less sure what you want to do?On 2025. May 31., at 10:59, zzlzef ***@***.***> wrote:﻿zzlzef left a comment (ssolo/ALE#42)

Hi, that is not a software question, I don't know what the research question is or what hypothesis you are trying to test.

Hi, thanks. ALE reconciles a sample of gene trees with a species tree. I’m analyzing multiple species, and each species includes many genomes. when i construct the species tree, should I select only one representative genome per species to construct the species tree?

**Comment by zzlzef** [comment]
> Hi, yes, I know what ALE does. I'm less sure what you want to do?On 2025. May 31., at 10:59, zzlzef ***@***.***> wrote:﻿zzlzef left a comment ([ssolo/ALE#42](https://github.com/ssolo/ALE/issues/42))
> 
> Hi, that is not a software question, I don't know what the research question is or what hypothesis you are trying to test.
> 
> Hi, thanks. ALE reconciles a sample of gene trees with a species tree. I’m analyzing multiple species, and each species includes many genomes. when i construct the species tree, should I select only one representative genome per species to construct the species tree?
> 
> —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
HI, I'm sorry I didn't describe my questions clearly. my question is that, i have a speices tree that includes same species (name) at the tree leaf, Can Ale use such trees for analysis?

**Comment by ssolo** [comment]
No. Leaf names in the species tree have to be unique.On 2025. May 31., at 11:18, zzlzef ***@***.***> wrote:﻿zzlzef left a comment (ssolo/ALE#42)

Hi, yes, I know what ALE does. I'm less sure what you want to do?On 2025. May 31., at 10:59, zzlzef @.***> wrote:﻿zzlzef left a comment (ssolo/ALE#42)
Hi, that is not a software question, I don't know what the research question is or what hypothesis you are trying to test.
Hi, thanks. ALE reconciles a sample of gene trees with a species tree. I’m analyzing multiple species, and each species includes many genomes. when i construct the species tree, should I select only one representative genome per species to construct the species tree?
—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: @.***>
HI, I'm sorry I didn't describe my questions clearly. my question is that, i have a speices tree that includes same species (name) at the tree leaf, Can Ale use such trees for analysis?

**Comment by zzlzef** [comment]
> No. Leaf names in the species tree have to be unique.On 2025. May 31., at 11:18, zzlzef ***@***.***> wrote:﻿zzlzef left a comment ([ssolo/ALE#42](https://github.com/ssolo/ALE/issues/42))
> 
> Hi, yes, I know what ALE does. I'm less sure what you want to do?On 2025. May 31., at 10:59, zzlzef @.***> wrote:﻿zzlzef left a comment ([ssolo/ALE#42](https://github.com/ssolo/ALE/issues/42))
> Hi, that is not a software question, I don't know what the research question is or what hypothesis you are trying to test.
> Hi, thanks. ALE reconciles a sample of gene trees with a species tree. I’m analyzing multiple species, and each species includes many genomes. when i construct the species tree, should I select only one representative genome per species to construct the species tree?
> —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: @.***>
> HI, I'm sorry I didn't describe my questions clearly. my question is that, i have a speices tree that includes same species (name) at the tree leaf, Can Ale use such trees for analysis?
> 
> 
> —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

Ok, thanks a lot. and I'm very sorry to bother you. I have read many of your published articles (e.g. new tool AleRax, and the review article about Phylogenetic reconciliation published in ISME) and I am impressed by the outstanding work you have done. I'm very glad to have been able to get in touch with you by chance today.

**Comment by zzlzef** [comment]
> No. Leaf names in the species tree have to be unique.On 2025. May 31., at 11:18, zzlzef ***@***.***> wrote:﻿zzlzef left a comment ([ssolo/ALE#42](https://github.com/ssolo/ALE/issues/42))
> 
> Hi, yes, I know what ALE does. I'm less sure what you want to do?On 2025. May 31., at 10:59, zzlzef @.***> wrote:﻿zzlzef left a comment ([ssolo/ALE#42](https://github.com/ssolo/ALE/issues/42))
> Hi, that is not a software question, I don't know what the research question is or what hypothesis you are trying to test.
> Hi, thanks. ALE reconciles a sample of gene trees with a species tree. I’m analyzing multiple species, and each species includes many genomes. when i construct the species tree, should I select only one representative genome per species to construct the species tree?
> —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: @.***>
> HI, I'm sorry I didn't describe my questions clearly. my question is that, i have a speices tree that includes same species (name) at the tree leaf, Can Ale use such trees for analysis?
> 
> 
> —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

hi, Gergely, another question about the gene tree, i'm also trying to use AleRax, as describtion in the review article about Phylogenetic reconciliation published in ISME, we need to cluster the sequences into gene families and infer trees for each gene family, so ,when a species have mutiple genomes, should i cluster all genes from multiple genomes of a species into gene families?

**Comment by zzlzef** [comment]
hi, Gergely, another question about the gene tree, i'm also trying to use AleRax, as describtion in the review article about Phylogenetic reconciliation published in ISME, we need to cluster the sequences into gene families and infer trees for each gene family, so ,when a species have mutiple genomes, should i cluster all genes from multiple genomes of a species into gene families?

> hi, Gergely, Benoit has solved my confusion. thanks again.
